kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -636,7 +636,6 @@ private void addFiltersToPipeline(
                             sniHostname,
                             virtualClusterModel,
                             inboundChannel,
-                            clientSaslManager,
```

Comment:
Why did you move away from the `clientSaslManager` being a field of the FilterHandler and into being a property of the channel. It feels like a completely different usage pattern which makes unit testing harder and requires the use of static methods to bind it to and recover it from the channel.